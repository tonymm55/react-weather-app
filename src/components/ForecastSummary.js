import React from "react";
import iconData from "../data/iconData.json";

// forecast summary component
function ForecastSummary({
  date, icon, temperature, description, onSelect,
}) {
  // const { date, icon, temperature, description, onSelect } = props;

  const weatherCode = `${icon.slice(0, 1)}00`;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary__date">{formattedDate}</div>
      <div className="forecast-summary__icon" data-testid="forecast-icon">
        <img src={iconData[weatherCode]} alt="icon" />
      </div>
      <div className="forecast-summary__temperature">
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-summary__description">{description}</div>
      <button type="button" onClick={() => onSelect(date)}>
        Click for details
      </button>
    </div>
  );
}

export default ForecastSummary;
