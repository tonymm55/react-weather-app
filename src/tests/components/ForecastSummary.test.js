import React from "react";
import { getAllByDisplayValue, getQueriesForElement, render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1111111,
    description: "Stub description",
    icon: "stubIcon",
    temperature: {
      min: 12,
      max: 22,
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getAllByText, getByTestId } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );

    expect(
      getAllByText(
        (content, element) =>
          element.classList.contains("forecast-summary__date") &&
          content.startsWith("Thu Jan 01")
      )[0]
    ).toBeInTheDocument();

    expect(getAllByText("Stub description")[0]).toHaveClass(
      "forecast-summary__description"
    );
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary__icon");
    expect(getAllByText("22Â°C")[0]).toHaveClass("forecast-summary__temperature");
    expect(getAllByText("Thu Jan 01 1970")[0]).toHaveClass("forecast-summary__date");
  });
});