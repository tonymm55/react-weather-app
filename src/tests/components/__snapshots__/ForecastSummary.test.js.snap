// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <img
        alt="icon"
      />
    </div>
    <div
      class="forecast-summary__temperature"
    >
      22Â°C
    </div>
    <div
      class="forecast-summary__description"
    >
      Stub description
    </div>
    <button
      type="button"
    >
      More Details
    </button>
  </div>
</DocumentFragment>
`;
